profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/binder/BaseProfitBricksRequestBinder.java

```diff
@@ -16,24 +16,24 @@
  */
 package org.apache.jclouds.profitbricks.rest.binder;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
 import java.util.Map;
-
 import org.jclouds.http.HttpRequest;
 import org.jclouds.rest.MapBinder;
-
 import com.google.common.base.Strings;
-
 import org.jclouds.io.MutableContentMetadata;
 import org.jclouds.io.payloads.BaseMutableContentMetadata;
+import java.net.URI;
+import static com.google.common.base.Preconditions.checkNotNull;
+import com.google.common.base.Supplier;
 
 public abstract class BaseProfitBricksRequestBinder<T> implements MapBinder {
 
+   protected Supplier<URI> endpointSupplier;
```

Comment:
This should be declared final and added as a parameter to the constructor. You want to make sure this is always set by subclasses, ottherwise NPEs might appear (there is no need to annotate with `@Inject` or `@Provider` here; the way the supplier is used is up to the subclasses).